app/models/challenge.rb

```diff
@@ -32,7 +32,8 @@ class Challenge < ApplicationRecord
     elsif user.director?
       with_users_and_spots.where(users: { instrument: user.instrument })
     elsif user.squad_leader?
-      with_users_and_spots.where(spots: { row: user.spot.row })
+      ids = with_users_and_spots.where(spots: { row: user.spot.row }).pluck(:id)
+      with_users_and_spots.where(id: ids)
```

Comment:
I can't think of a better way to do this in one query right now, but I'll think on it a tad.

It'd likely have to be without ActiveRecord and probably a sub-query

Reply:
I'm fine with 2 queries so long as it's not N+1, we're good :)